service/src/main/java/com/commercetools/pspadapter/payone/domain/payone/PayonePostServiceImpl.java

```diff
@@ -58,6 +68,43 @@ public static PayonePostServiceImpl of(final String payoneServerApiUrl) throws I
         }
     }
 
+    /**
+     * Expand entries with list values to key-value pairs where the keys are transformed to set of {@code key[i]} with
+     * respective values from the list. Non-list arguments remain the same.
+     * <p>
+     * The indices origin is <b>1</b>.
+     * <p>
+     * <b>Note:</b> so far only {@link List} values are expanded, arrays behavior is still undefined.
+     * <p>
+     * Example:
+     * <pre> {"a": [5, 7], "b": "c", "d": ["e", "f"]} =>
+     * {"a[1]": 5, "a[2]": 7, "b": "c", "d[1]": "e", "d[2]": "f"}
+     * </pre>
+     *
+     * @param parameters Set of key-value pairs to expand
+     * @return new {@link Map} where keys with list values
+     */
+    @Nonnull
+    Map<String, Object> getObjectMapWithExpandedLists(Map<String, Object> parameters) {
```

Comment:
`BaseRequest` is a POJO bean, it doesn't have dynamic properties with names like `a[1]`, but it could have array/list properties instead. The [`BaseRequest#toStringMap`](https://github.com/commercetools/commercetools-payone-integration/blob/58bd98cb85012e6dd0a34b3b276590e435b92efb/service/src/main/java/com/commercetools/pspadapter/payone/domain/payone/model/common/BaseRequest.java#L82-L82) can convert the bean object to map of key=value pairs, where _key_ will be a list name, _value_ - `List` itself.

Unfortunately, lists format is not specified in HTTP/W3C standards, so every service could treat them as they want. By default `Unirest.post(Map)` will represent a list as: `a=[x, y , z]` (e.g. one value with 3 elements), whereas Payone requires them in format `a[1]=x&a[2]=y&a[3]=z` (e.g. 3 values).

That's why we should convert default Map representation to Payone specific request format.